storage/include/s3/metrics.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <Metrics.hpp>
+
+namespace concord::storage::s3 {
+
+struct Metrics {
+  Metrics()
+      : metrics_component_{concordMetrics::Component("s3", std::make_shared<concordMetrics::Aggregator>())},
+        numKeysTransferred{metrics_component_.RegisterCounter("keys_transferred")},
+        bytesTransferred{
+            metrics_component_.RegisterCounter("bytes_transferred"),
+        },
+        lastSavedBlockId{
+            metrics_component_.RegisterStatus("last_saved_block_id", "0"),
+        }
+
+  {
+    metrics_component_.Register();
+  }
+
+  void updateLastSavedBlockId(const std::string& key) {
+    if (!IsBlockKey(key)) return;
```

Comment:
AFAIK, the `key` must be the block id.
If not, can we at least log it as a warning?

Reply:
It can be the block id or metadata. At this stage we don't know it and we have to make a guess.